src/index.ts

```diff
@@ -34,14 +37,25 @@ interface Source {
 let validLanguages = ["vue"];
 
 export async function check(options: Options) {
-  const { workspace, onlyTemplate = false, onlyTypeScript = false, excludeDir } = options;
+  const { workspace, onlyTemplate = false, onlyTypeScript = false, excludeDir, files = [] } = options;
   if (onlyTypeScript) {
     validLanguages = ["ts", "tsx", "vue"];
   }
   const srcDir = options.srcDir || options.workspace;
   const excludeDirs = typeof excludeDir === "string" ? [excludeDir] : excludeDir;
-  const docs = await traverse(srcDir, onlyTypeScript, excludeDirs);
-
+  let docs: TextDocument[] = [];
+  if (files.length) {
+    docs.push(...files.filter((file) => path.extname(file) === ".vue")
+      .map((file) => TextDocument.create(
+        `file://${file}`,
+        "vue",
+        0,
+        fs.readFileSync(file, "utf8")
```

Comment:
It would probably be better to read the file asynchronously:

```suggestion
        await fs.readFile(file, "utf8")
```

Also, this block should better be extracted to an async function `enumerate`, just like `traverse`.